python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -217,18 +256,54 @@ def execute(self, settings: CLISettings) -> None:
 
 
 def _match_file(
-    path: pathlib.Path, s_type: t.Type[SignalType], index: SignalTypeIndex
-) -> t.Sequence[IndexMatch]:
+    path: pathlib.Path,
+    s_type: t.Type[SignalType],
+    index: SignalTypeIndex,
+    rotations: bool = False,
+) -> t.Sequence[_IndexMatchWithRotation]:
     if issubclass(s_type, MatchesStr):
-        return index.query(path.read_text())
+        matches = index.query(path.read_text())
+        return [_IndexMatchWithRotation(match=match) for match in matches]
+
     assert issubclass(s_type, FileHasher)
-    return index.query(s_type.hash_from_file(path))
+
+    if not rotations or s_type != PhotoContent:
+        matches = index.query(s_type.hash_from_file(path))
+        return [_IndexMatchWithRotation(match=match) for match in matches]
+
+    # Handle rotations for photos
+    with open(path, "rb") as f:
+        image_data = f.read()
+
+    rotated_images: t.Dict[RotationType, bytes] = PhotoContent.all_simple_rotations(
+        image_data
+    )
+    all_matches = []
+
+    for rotation_type, rotated_bytes in rotated_images.items():
+        # Create a temporary file to hold the image bytes
+        with tempfile.NamedTemporaryFile() as temp_file:
```

Comment:
blocking: This is going to write a lot of files over the course of an execution! We are going to call this method for every single match type.

Another approach you could do is refactor this so that the rotated images are inserted higher up in the stack, and then rather than taking rotations: bool, you could pass in the path of the rotation, and an optional enum representing which enum it can take.

Reply:
I'm thinking of change s_type to be subclass of BytesHasher so that I can use bytes directly with `hash_from_bytes`. All the tests passed and mypy doesn't complain. What do you think of this approach?

Reply:
In the last PR, I mentioned the downside - not all photo formats are knowable without the extension. There's likely a workaround, but let's stay with the current course and see if we can fix it in a followup instead.